src/webui/webui.c

```diff
@@ -410,11 +428,17 @@ http_stream_run(http_connection_t *hc, profile_chain_t *prch,
       break;
 
     case SMT_GRACE:
-      grace = sm->sm_code < 5 ? 5 : grace;
+      if (sm->sm_code > grace) {
```

Comment:
I don't quite understand the old logic. Grace period could be changed to 5, or left as it was.

In my system, every new stream ends up with `SM_GRACE` with a code of 3 being invoked. The value is upgraded to 5, and that downgrades the grace period of 20 to 5. So the actual grace period is always 5 seconds and not the hardcoded period of 20.

I am not sure that was intentional. I believe it was a bug. I have changed the behaviour so the grace period can only be increased, and to a minimum of 5 seconds. That is the best assumption I could make of the original intention and it was needed to make the code respect the timeouts.

This can be reworked if I misunderstood how it is supposed to work.

Reply:
I will see if I can make any sense of the old code or understand why it was done this way